### PR TITLE
fix(Dialog): keep Dialog open when press click on content and release on overlay

### PR DESCRIPTION
**Pull request by YuanboXue-Amber**
## Current Behavior
When a Dialog is opened, press click on content and release outside of Dialog will close the Dialog.

## New Behavior
Press click on Dialog content and release outside of it will not close the Dialog.

Fixes:
According to [click event definition](https://developer.mozilla.org/en-US/docs/Web/API/Element/click_event):
>If the button is pressed on one element and the pointer is moved outside the element before the button is released, the event is fired on the most specific ancestor element that contained both elements.

press mouse on dialog content and release outside will cause a click event to be fired on dialog overlay.
Events fired during this process: mousedown on content -> mousemove -> mouseup on dialog overlay -> click on dialog overlay.
The fix is to use a ref to register mousedown on content, and on click event handler, check this ref to decide if dialog should be closed.

e2e tests was added

### CODE REVIEW

**Comment by fabricteam** [comment]
## 📊 Bundle size report

<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/00a4a1dd582f5695b7cb9bba48fb8537534bbec0'>00a4a1dd582f5695b7cb9bba48fb8537534bbec0</a></sub>
<div id="bundleSizeComment9423"/>

**Comment by fabricteam** [comment]
## Perf Analysis (`@fluentui/react-northstar`)
:warning: 1 potential perf regressions detected

### Potential regressions comparing to master

Scenario | Current PR Ticks | Baseline Ticks | Ratio | Regression Analysis
:--- | ---:| ---:| ---: | ---: 
AlertMinimalPerf.default | [226](https://fluentuipr.z22.web.core.windows.net/pull/22849/perf-test-northstar/AlertMinimalPerf.default.html) | [198](https://fluentuipr.z22.web.core.windows.net/pull/22849/perf-test-northstar/AlertMinimalPerf.default_base.html) | 1.14:1 | [analysis](https://fluentuipr.z22.web.core.windows.net/pull/22849/perf-test-northstar/AlertMinimalPerf.default.regression.txt)
<details><summary>Perf tests with no regressions</summary>

Scenario | Current PR Ticks | Baseline Ticks | Ratio
:--- | ---:| ---:| ---:
ButtonMinimalPerf.default | [142](https://fluentuipr.z22.web.core.windows.net/pull/22849/perf-test-northstar/ButtonMinimalPerf.default.html) | [112](https://fluentuipr.z22.web.core.windows.net/pull/22849/perf-test-northstar/ButtonMinimalPerf.default_base.html) | 1.27:1
LoaderMinimalPerf.default | [589](https://fluentuipr.z22.web.core.windows.net/pull/22849/perf-test-northstar/LoaderMinimalPerf.default.html) | [510](https://fluentuipr.z22.web.core.windows.net/pull/22849/perf-test-northstar/LoaderMinimalPerf.default_base.html) | 1.15:1
PopupMinimalPerf.default | [541](https://fluentuipr.z22.web.core.windows.net/pull/22849/perf-test-northstar/PopupMinimalPerf.default.html) | [480](https://fluentuipr.z22.web.core.windows.net/pull/22849/perf-test-northstar/PopupMinimalPerf.default_base.html) | 1.13:1
MenuButtonMinimalPerf.default | [1429](https://fluentuipr.z22.web.core.windows.net/pull/22849/perf-test-northstar/MenuButtonMinimalPerf.default.html) | [1313](https://fluentuipr.z22.web.core.windows.net/pull/22849/perf-test-northstar/MenuButtonMinimalPerf.default_base.html) | 1.09:1
VideoMinimalPerf.default | [556](https://fluentuipr.z22.web.core.windows.net/pull/22849/perf-test-northstar/VideoMinimalPerf.default.html) | [521](https://fluentuipr.z22.web.core.windows.net/pull/22849/perf-test-northstar/VideoMinimalPerf.default_base.html) | 1.07:1
DividerMinimalPerf.default | [300](https://fluentuipr.z22.web.core.windows.net/pull/22849/perf-test-northstar/DividerMinimalPerf.default.html) | [283](https://fluentuipr.z22.web.core.windows.net/pull/22849/perf-test-northstar/DividerMinimalPerf.default_base.html) | 1.06:1
TreeWith60ListItems.default | [144](https://fluentuipr.z22.web.core.windows.net/pull/22849/perf-test-northstar/TreeWith60ListItems.default.html) | [136](https://fluentuipr.z22.web.core.windows.net/pull/22849/perf-test-northstar/TreeWith60ListItems.default_base.html) | 1.06:1
InputMinimalPerf.default | [1090](https://fluentuipr.z22.web.core.windows.net/pull/22849/perf-test-northstar/InputMinimalPerf.default.html) | [1036](https://fluentuipr.z22.web.core.windows.net/pull/22849/perf-test-northstar/InputMinimalPerf.default_base.html) | 1.05:1
ListWith60ListItems.default | [559](https://fluentuipr.z22.web.core.windows.net/pull/22849/perf-test-northstar/ListWith60ListItems.default.html) | [533](https://fluentuipr.z22.web.core.windows.net/pull/22849/perf-test-northstar/ListWith60ListItems.default_base.html) | 1.05:1
HeaderMinimalPerf.default | [298](https://fluentuipr.z22.web.core.windows.net/pull/22849/perf-test-northstar/HeaderMinimalPerf.default.html) | [286](https://fluentuipr.z22.web.core.windows.net/pull/22849/perf-test-northstar/HeaderMinimalPerf.default_base.html) | 1.04:1
MenuMinimalPerf.default | [708](https://fluentuipr.z22.web.core.windows.net/pull/22849/perf-test-northstar/MenuMinimalPerf.default.html) | [689](https://fluentuipr.z22.web.core.windows.net/pull/22849/perf-test-northstar/MenuMinimalPerf.default_base.html) | 1.03:1
RadioGroupMinimalPerf.default | [368](https://fluentuipr.z22.web.core.windows.net/pull/22849/perf-test-northstar/RadioGroupMinimalPerf.default.html) | [357](https://fluentuipr.z22.web.core.windows.net/pull/22849/perf-test-northstar/RadioGroupMinimalPerf.default_base.html) | 1.03:1
TableMinimalPerf.default | [338](https://fluentuipr.z22.web.core.windows.net/pull/22849/perf-test-northstar/TableMinimalPerf.default.html) | [328](https://fluentuipr.z22.web.core.windows.net/pull/22849/perf-test-northstar/TableMinimalPerf.default_base.html) | 1.03:1
TreeMinimalPerf.default | [684](https://fluentuipr.z22.web.core.windows.net/pull/22849/perf-test-northstar/TreeMinimalPerf.default.html) | [664](https://fluentuipr.z22.web.core.windows.net/pull/22849/perf-test-northstar/TreeMinimalPerf.default_base.html) | 1.03:1
AnimationMinimalPerf.default | [451](https://fluentuipr.z22.web.core.windows.net/pull/22849/perf-test-northstar/AnimationMinimalPerf.default.html) | [441](https://fluentuipr.z22.web.core.windows.net/pull/22849/perf-test-northstar/AnimationMinimalPerf.default_base.html) | 1.02:1
FormMinimalPerf.default | [337](https://fluentuipr.z22.web.core.windows.net/pull/22849/perf-test-northstar/FormMinimalPerf.default.html) | [331](https://fluentuipr.z22.web.core.windows.net/pull/22849/perf-test-northstar/FormMinimalPerf.default_base.html) | 1.02:1
GridMinimalPerf.default | [253](https://fluentuipr.z22.web.core.windows.net/pull/22849/perf-test-northstar/GridMinimalPerf.default.html) | [247](https://fluentuipr.z22.web.core.windows.net/pull/22849/perf-test-northstar/GridMinimalPerf.default_base.html) | 1.02:1
LayoutMinimalPerf.default | [300](https://fluentuipr.z22.web.core.windows.net/pull/22849/perf-test-northstar/LayoutMinimalPerf.default.html) | [293](https://fluentuipr.z22.web.core.windows.net/pull/22849/perf-test-northstar/LayoutMinimalPerf.default_base.html) | 1.02:1
RefMinimalPerf.default | [202](https://fluentuipr.z22.web.core.windows.net/pull/22849/perf-test-northstar/RefMinimalPerf.default.html) | [198](https://fluentuipr.z22.web.core.windows.net/pull/22849/perf-test-northstar/RefMinimalPerf.default_base.html) | 1.02:1
StatusMinimalPerf.default | [564](https://fluentuipr.z22.web.core.windows.net/pull/22849/perf-test-northstar/StatusMinimalPerf.default.html) | [553](https://fluentuipr.z22.web.core.windows.net/pull/22849/perf-test-northstar/StatusMinimalPerf.default_base.html) | 1.02:1
TextAreaMinimalPerf.default | [403](https://fluentuipr.z22.web.core.windows.net/pull/22849/perf-test-northstar/TextAreaMinimalPerf.default.html) | [396](https://fluentuipr.z22.web.core.windows.net/pull/22849/perf-test-northstar/TextAreaMinimalPerf.default_base.html) | 1.02:1
AttachmentMinimalPerf.default | [125](https://fluentuipr.z22.web.core.windows.net/pull/22849/perf-test-northstar/AttachmentMinimalPerf.default.html) | [124](https://fluentuipr.z22.web.core.windows.net/pull/22849/perf-test-northstar/AttachmentMinimalPerf.default_base.html) | 1.01:1
CheckboxMinimalPerf.default | [2282](https://fluentuipr.z22.web.core.windows.net/pull/22849/perf-test-northstar/CheckboxMinimalPerf.default.html) | [2261](https://fluentuipr.z22.web.core.windows.net/pull/22849/perf-test-northstar/CheckboxMinimalPerf.default_base.html) | 1.01:1
EmbedMinimalPerf.default | [3473](https://fluentuipr.z22.web.core.windows.net/pull/22849/perf-test-northstar/EmbedMinimalPerf.default.html) | [3438](https://fluentuipr.z22.web.core.windows.net/pull/22849/perf-test-northstar/EmbedMinimalPerf.default_base.html) | 1.01:1
RosterPerf.default | [933](https://fluentuipr.z22.web.core.windows.net/pull/22849/perf-test-northstar/RosterPerf.default.html) | [927](https://fluentuipr.z22.web.core.windows.net/pull/22849/perf-test-northstar/RosterPerf.default_base.html) | 1.01:1
SkeletonMinimalPerf.default | [288](https://fluentuipr.z22.web.core.windows.net/pull/22849/perf-test-northstar/SkeletonMinimalPerf.default.html) | [284](https://fluentuipr.z22.web.core.windows.net/pull/22849/perf-test-northstar/SkeletonMinimalPerf.default_base.html) | 1.01:1
TextMinimalPerf.default | [281](https://fluentuipr.z22.web.core.windows.net/pull/22849/perf-test-northstar/TextMinimalPerf.default.html) | [277](https://fluentuipr.z22.web.core.windows.net/pull/22849/perf-test-northstar/TextMinimalPerf.default_base.html) | 1.01:1
TooltipMinimalPerf.default | [907](https://fluentuipr.z22.web.core.windows.net/pull/22849/perf-test-northstar/TooltipMinimalPerf.default.html) | [895](https://fluentuipr.z22.web.core.windows.net/pull/22849/perf-test-northstar/TooltipMinimalPerf.default_base.html) | 1.01:1
ButtonOverridesMissPerf.default | [1268](https://fluentuipr.z22.web.core.windows.net/pull/22849/perf-test-northstar/ButtonOverridesMissPerf.default.html) | [1265](https://fluentuipr.z22.web.core.windows.net/pull/22849/perf-test-northstar/ButtonOverridesMissPerf.default_base.html) | 1:1
ButtonSlotsPerf.default | [445](https://fluentuipr.z22.web.core.windows.net/pull/22849/perf-test-northstar/ButtonSlotsPerf.default.html) | [445](https://fluentuipr.z22.web.core.windows.net/pull/22849/perf-test-northstar/ButtonSlotsPerf.default_base.html) | 1:1
ImageMinimalPerf.default | [298](https://fluentuipr.z22.web.core.windows.net/pull/22849/perf-test-northstar/ImageMinimalPerf.default.html) | [298](https://fluentuipr.z22.web.core.windows.net/pull/22849/perf-test-northstar/ImageMinimalPerf.default_base.html) | 1:1
SegmentMinimalPerf.default | [284](https://fluentuipr.z22.web.core.windows.net/pull/22849/perf-test-northstar/SegmentMinimalPerf.default.html) | [284](https://fluentuipr.z22.web.core.windows.net/pull/22849/perf-test-northstar/SegmentMinimalPerf.default_base.html) | 1:1
SliderMinimalPerf.default | [1412](https://fluentuipr.z22.web.core.windows.net/pull/22849/perf-test-northstar/SliderMinimalPerf.default.html) | [1407](https://fluentuipr.z22.web.core.windows.net/pull/22849/perf-test-northstar/SliderMinimalPerf.default_base.html) | 1:1
SplitButtonMinimalPerf.default | [3704](https://fluentuipr.z22.web.core.windows.net/pull/22849/perf-test-northstar/SplitButtonMinimalPerf.default.html) | [3701](https://fluentuipr.z22.web.core.windows.net/pull/22849/perf-test-northstar/SplitButtonMinimalPerf.default_base.html) | 1:1
IconMinimalPerf.default | [515](https://fluentuipr.z22.web.core.windows.net/pull/22849/perf-test-northstar/IconMinimalPerf.default.html) | [513](https://fluentuipr.z22.web.core.windows.net/pull/22849/perf-test-northstar/IconMinimalPerf.default_base.html) | 1:1
TableManyItemsPerf.default | [1603](https://fluentuipr.z22.web.core.windows.net/pull/22849/perf-test-northstar/TableManyItemsPerf.default.html) | [1598](https://fluentuipr.z22.web.core.windows.net/pull/22849/perf-test-northstar/TableManyItemsPerf.default_base.html) | 1:1
CustomToolbarPrototype.default | [2318](https://fluentuipr.z22.web.core.windows.net/pull/22849/perf-test-northstar/CustomToolbarPrototype.default.html) | [2310](https://fluentuipr.z22.web.core.windows.net/pull/22849/perf-test-northstar/CustomToolbarPrototype.default_base.html) | 1:1
ToolbarMinimalPerf.default | [795](https://fluentuipr.z22.web.core.windows.net/pull/22849/perf-test-northstar/ToolbarMinimalPerf.default.html) | [797](https://fluentuipr.z22.web.core.windows.net/pull/22849/perf-test-northstar/ToolbarMinimalPerf.default_base.html) | 1:1
AttachmentSlotsPerf.default | [863](https://fluentuipr.z22.web.core.windows.net/pull/22849/perf-test-northstar/AttachmentSlotsPerf.default.html) | [874](https://fluentuipr.z22.web.core.windows.net/pull/22849/perf-test-northstar/AttachmentSlotsPerf.default_base.html) | 0.99:1
CarouselMinimalPerf.default | [387](https://fluentuipr.z22.web.core.windows.net/pull/22849/perf-test-northstar/CarouselMinimalPerf.default.html) | [389](https://fluentuipr.z22.web.core.windows.net/pull/22849/perf-test-northstar/CarouselMinimalPerf.default_base.html) | 0.99:1
ChatDuplicateMessagesPerf.default | [247](https://fluentuipr.z22.web.core.windows.net/pull/22849/perf-test-northstar/ChatDuplicateMessagesPerf.default.html) | [249](https://fluentuipr.z22.web.core.windows.net/pull/22849/perf-test-northstar/ChatDuplicateMessagesPerf.default_base.html) | 0.99:1
DatepickerMinimalPerf.default | [4919](https://fluentuipr.z22.web.core.windows.net/pull/22849/perf-test-northstar/DatepickerMinimalPerf.default.html) | [4945](https://fluentuipr.z22.web.core.windows.net/pull/22849/perf-test-northstar/DatepickerMinimalPerf.default_base.html) | 0.99:1
DialogMinimalPerf.default | [626](https://fluentuipr.z22.web.core.windows.net/pull/22849/perf-test-northstar/DialogMinimalPerf.default.html) | [633](https://fluentuipr.z22.web.core.windows.net/pull/22849/perf-test-northstar/DialogMinimalPerf.default_base.html) | 0.99:1
HeaderSlotsPerf.default | [616](https://fluentuipr.z22.web.core.windows.net/pull/22849/perf-test-northstar/HeaderSlotsPerf.default.html) | [625](https://fluentuipr.z22.web.core.windows.net/pull/22849/perf-test-northstar/HeaderSlotsPerf.default_base.html) | 0.99:1
ListCommonPerf.default | [534](https://fluentuipr.z22.web.core.windows.net/pull/22849/perf-test-northstar/ListCommonPerf.default.html) | [537](https://fluentuipr.z22.web.core.windows.net/pull/22849/perf-test-northstar/ListCommonPerf.default_base.html) | 0.99:1
ListMinimalPerf.default | [416](https://fluentuipr.z22.web.core.windows.net/pull/22849/perf-test-northstar/ListMinimalPerf.default.html) | [419](https://fluentuipr.z22.web.core.windows.net/pull/22849/perf-test-northstar/ListMinimalPerf.default_base.html) | 0.99:1
ProviderMergeThemesPerf.default | [1056](https://fluentuipr.z22.web.core.windows.net/pull/22849/perf-test-northstar/ProviderMergeThemesPerf.default.html) | [1071](https://fluentuipr.z22.web.core.windows.net/pull/22849/perf-test-northstar/ProviderMergeThemesPerf.default_base.html) | 0.99:1
DropdownMinimalPerf.default | [2506](https://fluentuipr.z22.web.core.windows.net/pull/22849/perf-test-northstar/DropdownMinimalPerf.default.html) | [2559](https://fluentuipr.z22.web.core.windows.net/pull/22849/perf-test-northstar/DropdownMinimalPerf.default_base.html) | 0.98:1
ItemLayoutMinimalPerf.default | [958](https://fluentuipr.z22.web.core.windows.net/pull/22849/perf-test-northstar/ItemLayoutMinimalPerf.default.html) | [983](https://fluentuipr.z22.web.core.windows.net/pull/22849/perf-test-northstar/ItemLayoutMinimalPerf.default_base.html) | 0.97:1
ChatWithPopoverPerf.default | [318](https://fluentuipr.z22.web.core.windows.net/pull/22849/perf-test-northstar/ChatWithPopoverPerf.default.html) | [332](https://fluentuipr.z22.web.core.windows.net/pull/22849/perf-test-northstar/ChatWithPopoverPerf.default_base.html) | 0.96:1
ChatMinimalPerf.default | [584](https://fluentuipr.z22.web.core.windows.net/pull/22849/perf-test-northstar/ChatMinimalPerf.default.html) | [615](https://fluentuipr.z22.web.core.windows.net/pull/22849/perf-test-northstar/ChatMinimalPerf.default_base.html) | 0.95:1
CardMinimalPerf.default | [426](https://fluentuipr.z22.web.core.windows.net/pull/22849/perf-test-northstar/CardMinimalPerf.default.html) | [451](https://fluentuipr.z22.web.core.windows.net/pull/22849/perf-test-northstar/CardMinimalPerf.default_base.html) | 0.94:1
ListNestedPerf.default | [432](https://fluentuipr.z22.web.core.windows.net/pull/22849/perf-test-northstar/ListNestedPerf.default.html) | [458](https://fluentuipr.z22.web.core.windows.net/pull/22849/perf-test-northstar/ListNestedPerf.default_base.html) | 0.94:1
AvatarMinimalPerf.default | [147](https://fluentuipr.z22.web.core.windows.net/pull/22849/perf-test-northstar/AvatarMinimalPerf.default.html) | [158](https://fluentuipr.z22.web.core.windows.net/pull/22849/perf-test-northstar/AvatarMinimalPerf.default_base.html) | 0.93:1
FlexMinimalPerf.default | [209](https://fluentuipr.z22.web.core.windows.net/pull/22849/perf-test-northstar/FlexMinimalPerf.default.html) | [227](https://fluentuipr.z22.web.core.windows.net/pull/22849/perf-test-northstar/FlexMinimalPerf.default_base.html) | 0.92:1
PortalMinimalPerf.default | [131](https://fluentuipr.z22.web.core.windows.net/pull/22849/perf-test-northstar/PortalMinimalPerf.default.html) | [142](https://fluentuipr.z22.web.core.windows.net/pull/22849/perf-test-northstar/PortalMinimalPerf.default_base.html) | 0.92:1
ReactionMinimalPerf.default | [269](https://fluentuipr.z22.web.core.windows.net/pull/22849/perf-test-northstar/ReactionMinimalPerf.default.html) | [293](https://fluentuipr.z22.web.core.windows.net/pull/22849/perf-test-northstar/ReactionMinimalPerf.default_base.html) | 0.92:1
BoxMinimalPerf.default | [252](https://fluentuipr.z22.web.core.windows.net/pull/22849/perf-test-northstar/BoxMinimalPerf.default.html) | [276](https://fluentuipr.z22.web.core.windows.net/pull/22849/perf-test-northstar/BoxMinimalPerf.default_base.html) | 0.91:1
LabelMinimalPerf.default | [293](https://fluentuipr.z22.web.core.windows.net/pull/22849/perf-test-northstar/LabelMinimalPerf.default.html) | [322](https://fluentuipr.z22.web.core.windows.net/pull/22849/perf-test-northstar/LabelMinimalPerf.default_base.html) | 0.91:1
AccordionMinimalPerf.default | [117](https://fluentuipr.z22.web.core.windows.net/pull/22849/perf-test-northstar/AccordionMinimalPerf.default.html) | [130](https://fluentuipr.z22.web.core.windows.net/pull/22849/perf-test-northstar/AccordionMinimalPerf.default_base.html) | 0.9:1
DropdownManyItemsPerf.default | [488](https://fluentuipr.z22.web.core.windows.net/pull/22849/perf-test-northstar/DropdownManyItemsPerf.default.html) | [553](https://fluentuipr.z22.web.core.windows.net/pull/22849/perf-test-northstar/DropdownManyItemsPerf.default_base.html) | 0.88:1
ProviderMinimalPerf.default | [298](https://fluentuipr.z22.web.core.windows.net/pull/22849/perf-test-northstar/ProviderMinimalPerf.default.html) | [338](https://fluentuipr.z22.web.core.windows.net/pull/22849/perf-test-northstar/ProviderMinimalPerf.default_base.html) | 0.88:1

</details><div id="perfComment9424"/>

File: packages/fluentui/react-northstar/src/components/Dialog/Dialog.tsx [hunk 1]
```diff
@@ -258,10 +258,24 @@ export const Dialog = (React.forwardRef<HTMLDivElement, DialogProps>((props, ref
     },
   });
 
+  // when press left click on Dialog content and hold, and mouse up on Dialog overlay, Dialog should keep open
+  const isMouseDownInsideContent = React.useRef(false);
```
Comment (ling1726) [inline packages/fluentui/react-northstar/src/components/Dialog/Dialog.tsx:262]: I never see where this ref is set to `false again`.

File: packages/fluentui/react-northstar/src/components/Dialog/Dialog.tsx [hunk 2]
```diff
@@ -258,10 +258,24 @@ export const Dialog = (React.forwardRef<HTMLDivElement, DialogProps>((props, ref
...
+  const registerMouseDownOnDialogContent = (e: React.MouseEvent) => {
```
Comment (ling1726) [inline packages/fluentui/react-northstar/src/components/Dialog/Dialog.tsx:263]: this should be testable in cypress by triggering a mousedown followed by a click somewhere else ?

Comment (YuanboXue-Amber) [reply packages/fluentui/react-northstar/src/components/Dialog/Dialog.tsx:263]: Heyy, if I do 
```
cy.trigger('mousedown').
trigger('mousemove', {clientX: 0, clientY: 0}). // move mouse to top left corner of page
trigger('mouseup')
```
It mimics the behavior of dragging mouse from content to outside of content. But unlike browser, cypress will not trigger an automatic click event after that. 

And if I chain it with a `.trigger('click')`, the test would be testing "clicking outside of dialog closes it". And it lost the purpose of testing the automatic click triggered after dragging.

Comment (YuanboXue-Amber) [reply packages/fluentui/react-northstar/src/components/Dialog/Dialog.tsx:263]: Ahaa it worked ❤️ on cypress with my change chained 'click' does not close dialog

**Review by ling1726 — APPROVED** [review]

**Pull request merged**